src/JsonRpc/Reciever.cs

```diff
@@ -77,7 +77,7 @@ protected virtual Renor GetRenor(JToken @object)
                 return new InvalidRequest(requestId, "Method not set");
             }
 
-            var hasParams = request.TryGetValue("params", out var @params);
+            var hasParams = request.TryGetValue("params", out var @params) && @params != null;
             if (hasParams && @params.Type != JTokenType.Array && @params.Type != JTokenType.Object)
```

Comment:
We should change this line to better align to the spec.  [`params` if specified must be an array or object](http://www.jsonrpc.org/specification#parameter_structures).

Swap with:
`if (hasParams && @params?.Type != JTokenType.Array && @params?.Type != JTokenType.Object)`

Reply:
Ah, ok, you want to use the `?.` operator. Sorry, fair enough - I missed that when viewing on my phone, I get you now. Will fix it up shortly :)